### PR TITLE
feat: cart managment

### PR DESCRIPTION
**Pull request by Anish-JS**
added cart management functionalities.
Please review.

Login using--> email: test@gmail.com, password: test123
Or you can signup using your own data

### CODE REVIEW

File: src/utils/cartServices/cartDetails.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  const discount = cartItems.reduce(
+    (acc, curr) =>
+      (acc +=
+        parseFloat(curr.discount) * parseInt(curr.price) * parseInt(curr.qty)),
+    0
+  );
+  const deliveryCharges = cartItems.reduce(
+    (acc, curr) => (acc += parseInt(curr.deliveryCharges)),
+    0
+  );
```
Comment (manismk) [inline src/utils/cartServices/cartDetails.js]: Suggestion: You can try converting it into one reduce method by passing acc as object {totalPrice:0,discount:0, deliveryCharges:0}

File: src/store/cart-context.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+const CartContext = createContext();
+
+const cartReducer = (state, action) => {
+  switch (action.type) {
+    case "SHOW_CART":
+      return { ...state, cartItems: action.payload.cart };
+
+    case "ADD":
+      console.log(action.payload.updatedCart);
```
Comment (manismk) [inline src/store/cart-context.js]: You can remove the console.log

File: src/components/Cart/Cart.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import CartCard from "../Card/CartCard";
+import "./Cart.css";
+import CartSummary from "./CartSummary";
+
+const Cart = () => {
+  const { isLoggedIn } = useAuth();
+  const { cartState } = useCart();
+
+  const { cartItems } = cartState;
+  console.log(cartItems);
```
Comment (manismk) [inline src/components/Cart/Cart.js]: You can remove the console log

File: src/utils/cartServices/changeQty.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import axios from "axios";
+
+const changeQty = async (id, dispatchCartState, changeType) => {
+  if (localStorage.getItem("token")) {
+    console.log(changeType);
```
Comment (manismk) [inline src/utils/cartServices/changeQty.js]: you can remove the console.log

File: src/components/Card/CartCard.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+const CartCard = (props) => {
+  const { cartState, dispatchCartState } = useCart();
+  const { addToWishListHandler, wishList } = useWishList();
+  const index = wishList.findIndex((item) => item["_id"] === props.data["_id"]);
+  const changeQtyHandler = (e) => {
+    let changeType;
+    const existingCartItem = cartState.cartItems.filter(
+      (item) => item["_id"] === props.data["_id"]
+    );
+    console.log(existingCartItem[0].qty);
```
Comment (manismk) [inline src/components/Card/CartCard.js]: you can remove the console log

File: src/components/Card/CartCard.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+      <div class="left-section-cart xs-padding xs-gutter sm-brr">
+        <img
+          src={props.data.image}
+          class="img-height-cart xs-padding"
+          alt="shoe-img"
+        />
+
+        <div class="product-details sm-gutter">
+          <div class="title para-md xs-gutter bold">{props.data.name}</div>
+          <div class="xs-gutter price-card">
```
Comment (manismk) [inline src/components/Card/CartCard.js]: change to className

File: src/components/Card/CartCard.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+          </button>
+          {index === -1 ? (
+            <button
+              class="primary-btn btn-style para-sm bold xs-gutter center"
+              onClick={() => addToWishListHandler(props.data)}
+            >
+              Add to wishlist
+            </button>
+          ) : (
+            <button class="primary-btn btn-style para-sm bold xs-gutter center">
```
Comment (manismk) [inline src/components/Card/CartCard.js]: change to className

File: src/components/Cart/Cart.js [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    <CartCard data={item} key={item["_id"]} />
+  ));
+
+  return (
+    <>
+      {!isLoggedIn ? (
+        <h3 className="center sm-gutter">Please Login</h3>
+      ) : cartItems.length === 0 ? (
+        <>
+          <h2 class="center xs-gutter">My Cart({cartItems.length})</h2>
```
Comment (manismk) [inline src/components/Cart/Cart.js]: change to className

File: src/components/Cart/Cart.js [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+          <h3 className="center xs-gutter">There no items here</h3>
+        </>
+      ) : (
+        <>
+          <h2 class="center xs-gutter">My Cart({cartItems.length})</h2>
+          <div class="section-cart sm-gutter">
+            <div className="cartItem-section">{cartItemList}</div>
+            {
+              <div class="right-section-cart xs-padding">
+                <CartSummary details={cartItems}></CartSummary>
```
Comment (manismk) [inline src/components/Cart/Cart.js]: change to className

File: src/components/Cart/CartSummary.js [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+    props.details
+  );
+  return (
+    <>
+      <div class="product-details sm-gutter">
+        <div class="title para-sm xs-gutter bold">Price Details</div>
+        <div class="price xs-gutter">
+          <div>Price ({props.details.length})</div>
+          <div>
+            <img
```
Comment (manismk) [inline src/components/Cart/CartSummary.js]: change to classname

File: src/components/Cart/CartSummary.js [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+              alt="rupee"
+              className="img-xxxs rupee-icon xxxs-padding"
+            />
+            {deliveryCharges}
+          </div>
+        </div>
+        <div class="price xs-gutter top-padding">
+          <div class="bold">Total Amount</div>
+          <div class="bold">
+            <img
```
Comment (manismk) [inline src/components/Cart/CartSummary.js]: change to className

**Review by manismk — COMMENTED** [review]
If I click Add to cart quickly twice duplicate items are added to the cart.
Gave some suggestions.

**Review by shraddha15-creator — APPROVED** [review]
The code looks good and is very well structured in the folder.

File: src/components/Cart/CartSummary.js (hunk 11 above)
Comment (Anish-JS) [reply src/components/Cart/CartSummary.js]: made all the changes please review and approve

**Review by manismk — APPROVED** [review]

**Pull request merged**